### PR TITLE
API Add document sets, remove *Page methods from DMSDocument

### PR DESCRIPTION
**Pull request by robbieaverill**
* Add 2.0.0 changelog
* Update DMSInterface and DMSDocumentInterface removing *page and adding getDocumentSetsByPage to DMSInterface
* Update use documentation and update unit tests

This commit changes the relationship from Pages has_many Documents to Pages has_many DocumentSets which are many_many to Documents. The upload field has been upated to attach documents to a set instead of a page, the tests updated and the DMSInterface and DMSDocumentInterface updated to be less relevant to pages and more relevant to document sets.

---

Migration strategy and/or build task for data migration is still to come.

Issue: #95

### CODE REVIEW

**Comment by robbieaverill** [comment]
Coverage is going down, I'll add some more tests

File: code/cms/DMSGridFieldDetailForm.php [hunk 1]
```diff
@@ -13,7 +13,7 @@ public function ItemEditForm()
 
         //add a data attribute specifying how many pages this document is referenced on
         if ($record = $this->record) {
-            $numberOfPageRelations = $record->Pages()->Count();
+            $numberOfPageRelations = $record->getRelatedPages()->Count();
```
Comment (fspringveldt) [inline code/cms/DMSGridFieldDetailForm.php]: This should be count() like the others.

File: docs/en/changelogs/2.0.0.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# 2.0.0 (unreleased)
+
+## Document sets
+
+Documents now belong to "sets", which are attached to Pages. A Page can have many Document Sets, and a Set has a
+many_many relationship with Documents.
+
+When upgrading from 1.x to 2.x you will need to migrate the relationships from your Pages to Documents so support
```
Comment (fspringveldt) [inline docs/en/changelogs/2.0.0.md]: Typo: from your Pages to Documents _so_ should be _to_

**Comment by robbieaverill** [comment]
Thanks @fspringveldt - I've fixed your comments. I think Sacha's got some UX suggestions for this to come.

**Pull request merged**